### PR TITLE
[web:a11y] make header a proper <header>

### PR DESCRIPTION
**Pull request by yjbanov**
Now that we have [proper headings](https://github.com/flutter/engine/blob/main/lib/web_ui/lib/src/engine/semantics/heading.dart), headers should become proper headers.

Fixes https://github.com/flutter/flutter/issues/152268

### CODE REVIEW

File: lib/web_ui/lib/src/engine/semantics/semantics.dart [hunk 1]
```diff
@@ -677,23 +681,18 @@ final class GenericRole extends SemanticRole {
       return;
     }
 
-    // Assign one of three roles to the element: group, heading, text.
+    // Assign one of three roles to the element: group or text.
```
Comment (navaronbracke) [inline lib/web_ui/lib/src/engine/semantics/semantics.dart:684]: This comment is now outdated?

```suggestion
    // Assign one of two roles to the element: group or text.
```

File: lib/web_ui/lib/src/engine/semantics/semantics.dart [hunk 2]
```diff
@@ -1261,11 +1260,24 @@ class SemanticsObject {
...
   /// Whether this object represents a heading element.
+  ///
+  /// Typically, a heading is a prominent piece of text that describes what the
+  /// rest of the screen or page is about.
+  ///
+  /// Not to be confused with [isHeader].
   bool get isHeading => headingLevel != 0;
 
-    /// Whether this object represents an editable text field.
+  /// Whether this object represents an editable text field.
```
Comment (navaronbracke) [inline lib/web_ui/lib/src/engine/semantics/semantics.dart:1270]:
```suggestion
  /// Whether this object represents an interactive link.
```

This doc seems wrong? This isn't talking about `ui.SemanticsFlag.isTextField`

**Comment by math1man** [comment]
Hey folks, this change broke our app. We're running into two issues.

1. The label of our page headers used to be read as "header, This is my page". Now the label is dropped and it's just read as "banner".
2. According to [documentation](http://go/mdn/Accessibility/ARIA/Roles/banner_role), banner is meant for 
    > defining a global site header, which usually includes a logo, company name, search feature, and possibly the global navigation or a slogan

    In our web app, we have a corresponding banner, which can logically be labeled as "banner", but the in-page headers that are impacted by this change should still be labeled as "header" rather than "banner.

Internal bug b/374695603.

**Comment by math1man** [comment]
Regarding my point #2, here's the issue using Wikipedia as a proxy, given that it has a very similar layout to our app: 
image

**Pull request merged**